prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,298 @@
+                < veh_range * const.ER
+            ):
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")] = (
+                    newdata.iloc[
+                        i, newdata.columns.get_loc("trip start battery charge")
+                    ]
+                    - newdata.iloc[i, newdata.columns.get_loc("Miles traveled")]
+                    * kwhmi
+                    * const.ER
+                )
+
+            else:
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 0
+
+            trip_data = newdata.iloc[i]
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging power")
+            ] = charging.get_charging_power(
+                power,
+                trip_strategy,
+                location_strategy,
+                kwh,
+                trip_data,
+            )
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging time")
+            ] = charging.get_charging_time(
+                newdata.iloc[i, newdata.columns.get_loc("charging power")],
+                kwh,
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")],
+                const.charging_efficiency,
+                trip_data,
+            )
+
+            newdata.iloc[i, newdata.columns.get_loc("charging consumption")] = (
+                newdata.iloc[i, newdata.columns.get_loc("charging power")]
+                * newdata.iloc[i, newdata.columns.get_loc("charging time")]
+                * const.charging_efficiency
+            )
+
+            start_point = round(
+                newdata.iloc[i, newdata.columns.get_loc("End time (hour decimal)")]
+                * 100
+            )
+            end_point = start_point + round(
+                newdata.iloc[i, newdata.columns.get_loc("charging time")] * 100
+            )
+            weekend_electricload[start_point:end_point] += newdata.iloc[
+                i, newdata.columns.get_loc("charging power")
+            ]
+
+    weekend_output_electricload = np.zeros(48)
+    for k in range(48):
+        if k == 0:
+            weekend_output_electricload[k] = sum(weekend_electricload[:100]) / 100
+        elif k == 47:
+            weekend_output_electricload[k] = sum(weekend_electricload[4700:]) / 100
+        else:
+            weekend_output_electricload[k] = (
+                sum(weekend_electricload[(k + 1) * 100 - 150 : (k + 1) * 100 - 50])
+                / 100
+            )
+
+    outputs = [weekend_output_electricload, weekday_output_electricload]
+
+    for day_iter in range(len(input_day)):
+        # weekday = 1 if weekday, 0 if not weekday
+        weekday = input_day[day_iter] - 1
+
+        if day_iter == len(input_day) - 1:
+            TRANS_charge[day_iter * 24 :] += outputs[weekday][:24].conj().T
```

Comment:
Is `conj` the complex conjugate? Does that make a difference here, where I think all of our values are purely real?